ecr/ref.go

```diff
@@ -55,6 +61,58 @@ func ParseRef(ref string) (ECRSpec, error) {
 	return parseARN(stripped)
 }
 
+// NewECRSpec takes an ECR image name and then constructs and returns an ECRSpec struct
+func NewECRSpec(input string) (ECRSpec, error) {
+	input = strings.TrimPrefix(input, "https://")
+
+	// Matching on account, region
+	err := errors.New("Invalid ECR image name")
+	matches := ecrRegex.FindStringSubmatch(input)
+	if len(matches) < 3 {
+		return ECRSpec{}, err
+	}
+	region := matches[2]
+	account := matches[1]
+
+	// Get the correct partition given its region
+	partition, found := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !found {
+		return ECRSpec{}, err
+	}
+
+	// Separate out <name>:<tag> for checking validity
+	tokens := strings.Split(input, "/")
```

Comment:
You probably want this as [`strings.SplitN`](https://golang.org/pkg/strings/#SplitN) since the part after the first `/` should remain together as the resulting `Repository`.  You should then be able to pass this to `parseARNResource` to get the `Object`.